### PR TITLE
Added support for mapping doc entities

### PR DESCRIPTION
**Pull request by overthetop**
The project now contains entities used by the apifest-doclet project to generate documentation based on documentation tags.

### CODE REVIEW

File: apifest-api/src/main/java/com/apifest/api/MappingDocumentation.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+@XmlRootElement(name = "mappings_docs")
+public class MappingDocumentation implements Serializable {
+
+    private static final long serialVersionUID = 8027603159873624488L;
+
+    @XmlAttribute(name = "version", required = true)
+    private String version;
+
+    @XmlElement(name = "endpoints", type = MappingEndpointDocumentation.class, required = true)
+    private List<MappingEndpointDocumentation> mappingEndpontDocumentation;
```
Comment (rossitsaborissova) [inline apifest-api/src/main/java/com/apifest/api/MappingDocumentation.java]: Please, fix the typo in the name of the variable - one "i" is missing in the endpoint part of the name.

File: apifest-api/src/main/java/com/apifest/api/MappingEndpointDocumentation.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2013-2014, ApiFest project
```
Comment (rossitsaborissova) [inline apifest-api/src/main/java/com/apifest/api/MappingEndpointDocumentation.java]: You can update the date to 2015.

File: apifest-api/src/main/java/com/apifest/api/MappingEndpointDocumentation.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlTransient;
+import javax.xml.bind.annotation.XmlType;
+
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlType(name = "endpoint_documentation")
+public class MappingEndpointDocumentation implements Serializable {
```
Comment (rossitsaborissova) [inline apifest-api/src/main/java/com/apifest/api/MappingEndpointDocumentation.java]: Please, add Javadoc to the class (in that case Javadoc at the class level only is enough).

File: apifest-api/src/main/java/com/apifest/api/MappingEndpointDocumentation.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlTransient;
+import javax.xml.bind.annotation.XmlType;
+
```
Comment (rossitsaborissova) [inline apifest-api/src/main/java/com/apifest/api/MappingEndpointDocumentation.java]: Please, add Javadoc to the class (in that case Javadoc at the class level only is enough).

File: apifest-api/src/main/java/com/apifest/api/MappingEndpointDocumentation.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    @XmlAttribute(name = "description", required = true)
+    private String description;
+
+    @XmlAttribute(name = "summary", required = true)
+    private String summary;
+
+    @XmlAttribute(name = "group", required = true)
+    private String group;
+
+    @XmlElement(name = "params", type = MappingEndpointParamDocumentation.class)
```
Comment (rossitsaborissova) [inline apifest-api/src/main/java/com/apifest/api/MappingEndpointDocumentation.java]: How these params are supposed to be used?

**Comment by overthetop** [comment]
I think that I have corrected all the things that you have noted. Can you take a look?

**Pull request merged**